File: GoLang/quotes/model/generalService.go
package model

import (
	"context"
	"time"

	pb "github.com/RYANCOAL9999/AI_Go_Proto_File"
	"github.com/gin-gonic/gin"
	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials/insecure"
)

type GeneralGRPCClient struct {
	client pb.GeneralServiceClient
}

func NewGeneralGRPCClient(addr string) (*GeneralGRPCClient, error) {
	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
	if err != nil {
		return nil, err
	}
	return &GeneralGRPCClient{
		client: pb.NewGeneralServiceClient(conn),
	}, nil
}

func InfoResponse(generalClient *GeneralGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "info"

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := generalClient.client.InfoEvent(ctx, &pb.InfoRequest{})

		if err != nil {
			HandleGRPCError(c, err, event_name)
			return
		}

		SendResponse(c, response, event_name)

	}

}

func DescriblerResponse(generalGRPCClient *GeneralGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "describle"

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := generalGRPCClient.client.DescriblerEvent(ctx, &pb.DescribeRequest{})

		if err != nil {
			HandleGRPCError(c, err, event_name)
			return
		}

		SendResponse(c, response, event_name)

	}

}

func HeaderResponse(generalGRPCClient *GeneralGRPCClient) gin.HandlerFunc {
	return func(c *gin.Context) {

		const event_name = "head"

		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
		defer cancel()

		response, err := generalGRPCClient.client.HeaderEvent(ctx, &pb.HeaderRequest{})

		if err != nil {
			HandleGRPCError(c, err, event_name)
			return
		}

		SendResponse(c, response, event_name)

	}
}

File: Python/quotes/model/generalService.py
import grpc

from dataclasses import asdict
from lib.proto.py.general_expression_pb2_grpc import GeneralServiceStub
from lib.proto.py.general_expression_pb2 import HeaderRequest, InfoRequest, DescribeRequest

class GeneralService(object):
    def __init__(
            self, 
            url:str
        )->None:
        self.__channel = grpc.insecure_channel(url)
        self.__stub = GeneralServiceStub(self.__channel)

    def headerResponse(self)->dict:
        with self.__channel:
            request_message = HeaderRequest()
            response = self.__stub.HeaderEvent(request_message)
            return asdict(response)
        
    def infoResponse(self)->dict:
        with self.__channel:
            request_message = InfoRequest()
            response = self.__stub.InfoEvent(request_message)
            return asdict(response)
        
    def describlerResponse(self)->dict:
        with self.__channel:
            request_message = DescribeRequest()
            response = self.__stub.DescriblerEvent(request_message)
            return asdict(response)

